crypto/src/keys/spend.rs

```diff
@@ -0,0 +1,123 @@
+use std::convert::{TryFrom, TryInto};
+
+use ark_ff::PrimeField;
+use decaf377::{self, FrExt};
+use rand_core::{CryptoRng, RngCore};
+
+use crate::Fr;
+
+use super::{
+    FullViewingKey, NullifierDerivingKey, NullifierPrivateKey, OutgoingViewingKey, ProofAuthKey,
+    SigningKey, SpendAuth, OVK_LEN_BYTES,
+};
+
+pub const SPENDSEED_LEN_BYTES: usize = 32;
+
+/// The root key material for a [`SpendKey`].
+#[derive(Clone, Debug)]
+pub struct SpendSeed(pub [u8; SPENDSEED_LEN_BYTES]);
+
+/// A key representing a single spending authority.
+pub struct SpendKey {
+    seed: SpendSeed,
+    ask: SigningKey<SpendAuth>,
+    fvk: FullViewingKey,
+    pak: ProofAuthKey,
+}
+
+impl SpendKey {
+    /// Generates a new [`SpendKey`] from the supplied RNG.
+    pub fn generate<R: RngCore + CryptoRng>(mut rng: R) -> Self {
+        let mut seed_bytes = [0u8; SPENDSEED_LEN_BYTES];
+        rng.fill_bytes(&mut seed_bytes);
+        Self::from_seed(SpendSeed(seed_bytes))
+    }
+
+    /// Loads a [`SpendKey`] from the provided [`SpendSeed`].
+    pub fn from_seed(seed: SpendSeed) -> Self {
+        let ask = {
+            let mut hasher = blake2b_simd::State::new();
+            hasher.update(b"Penumbra_ExpandSeed");
+            hasher.update(&seed.0);
+            hasher.update(&[0; 1]);
+            let hash_result = hasher.finalize();
+
+            // XXX: add Fr constructor to SigningKey
```

Comment:
We should add a constructor to the `SigningKey` API in `decaf377_rdsa` that takes a field element as input.